be/src/agent/agent_server.cpp

```diff
@@ -151,12 +151,23 @@ void AgentServer::Impl::init_or_die() {
         CHECK(st.ok()) << st;                                                            \
     } while (false)
 
-        // The ideal queue size of threadpool should be larger than the maximum number of tablet of a partition.
-        // But it seems that there's no limit for the number of tablets of a partition.
-        // Since a large queue size brings a little overhead, a big one is chosen here.
-        BUILD_DYNAMIC_TASK_THREAD_POOL("publish_version", config::transaction_publish_version_worker_count,
-                                       config::transaction_publish_version_worker_count,
-                                       DEFAULT_DYNAMIC_THREAD_POOL_QUEUE_SIZE, _thread_pool_publish_version);
+// The ideal queue size of threadpool should be larger than the maximum number of tablet of a partition.
+// But it seems that there's no limit for the number of tablets of a partition.
+// Since a large queue size brings a little overhead, a big one is chosen here.
+#ifdef BE_TEST
+        BUILD_DYNAMIC_TASK_THREAD_POOL("publish_version", 1, 1, DEFAULT_DYNAMIC_THREAD_POOL_QUEUE_SIZE,
+                                       _thread_pool_publish_version);
+#else
+        int max_publish_version_worker_count = config::transaction_publish_version_worker_count;
+        if (max_publish_version_worker_count <= 0) {
+            max_publish_version_worker_count = CpuInfo::num_cores() / 2;
```

Comment:
/2 ?

Reply:
When this merge, I will backport the next pr.